libgalois/src/analytics/bfs/bfs.cpp

```diff
@@ -390,21 +391,97 @@ SynchronousDirectOpt(
   delete next_frontier;
 }
 
+template <typename NDType, typename ValueTy>
+void
+InitializeNodeData(const ValueTy value, katana::LargeArray<NDType>* node_data) {
+  katana::do_all(katana::iterate(0ul, node_data->size()), [&](auto n) {
+    (*node_data)[n] = value;
+  });
+}
+
+template <typename NDType, typename ValueTy, typename... Args>
+void
+InitializeNodeData(
+    const ValueTy value, katana::LargeArray<NDType>* node_data, Args... args) {
+  InitializeNodeData(value, node_data);
+  InitializeNodeData(value, args...);
+}
+
+template <typename NDType>
+void
+InitializeGraphNodeData(
+    Graph* graph, const katana::LargeArray<NDType>& node_data) {
+  katana::do_all(katana::iterate(*graph), [&](auto& node) {
+    graph->GetData<BfsNodeParent>(node) = node_data[node];
+  });
+}
+
+void
+ComputeParentByDistance(
```

Comment:
This code is pretty much doing a pull-style BFS. Do we really want to implement Async BFS this way? A moment to step back and reflect (at least for me).

Reply:
Please rename to `ComputeParentFromDistance` or `ComputeParentUsingDistance`